packages/x-date-pickers/src/internals/components/CalendarOrClockPicker/CalendarOrClockPicker.tsx

```diff
@@ -148,6 +148,18 @@ export function CalendarOrClockPicker<TDate, View extends CalendarOrClockPickerV
     [isMobileKeyboardViewOpen, onViewChange, toggleMobileKeyboardView],
   );
 
+  if (process.env.NODE_ENV !== 'production') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const warnedOnceNotValidOpenTo = React.useRef(false);
```

Comment:
I think we can copy the `buildWarning` from the grid and us it.

Reply:
`buildWarning` method does not work well here, because `CalendarOrClockPicker` is rendered conditionally and `openTo` is changing.
I've experimented with several solutions and this seemed somewhat intrusive, but IMHO better than Alex's suggestion to have a variable up top, which would result in a single warning per app lifecycle, even though there might be a few pickers with incorrect props. 🤔

Reply:
Applied the fix suggested by Alex resulting in only a single warning being shown.

Reply:
> buildWarning method does not work well here, because CalendarOrClockPicker is rendered conditionally and openTo is changing and causing re-renders as well.

Not sure to understand the link between the two here